mkosi/machine.py

```diff
@@ -0,0 +1,97 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+from __future__ import annotations
+
+import contextlib
+import signal
+import sys
+from typing import Any, Optional, Sequence
+
+import pexpect  # type: ignore
+
+import mkosi
+
+from .backend import die
+
+
+class Machine():
+    def __init__(self, args: Optional[Sequence[str]] = None) -> None:
+        # Remains None until image is built and booted, then receives pexpect process
+        self.serial: pexpect.spawnu
+        self.stack = contextlib.ExitStack()
+        self.args = args
+        self.exit_code: int
+        self.verb: str
+
+
+    def __enter__(self) -> Machine:
+        with contextlib.ExitStack() as stack:
+            parsed_args = mkosi.parse_args(self.args)
+            parsed_args = parsed_args["default"]
+            loaded_args = mkosi.load_args(parsed_args)
+
+            self.verb = loaded_args.verb
+
+            mkosi.prepend_to_environ_path(loaded_args.extra_search_paths)
+
+            if loaded_args.verb in mkosi.MKOSI_COMMANDS_SUDO:
+                mkosi.check_root()
+                mkosi.unlink_output(loaded_args)
+
+            if loaded_args.verb == "build":
+                mkosi.check_output(loaded_args)
+
+            if mkosi.needs_build(loaded_args):
+                mkosi.check_root()
+                mkosi.check_native(loaded_args)
+                mkosi.init_namespace(loaded_args)
+                manifest = mkosi.build_stuff(loaded_args)
+
+                if loaded_args.auto_bump:
+                    mkosi.bump_image_version(loaded_args)
+
+                mkosi.save_manifest(loaded_args, manifest)
+
+                mkosi.print_output_size(loaded_args)
+
+            cmdline = None
```

Comment:
python variables are function scoped, not block scoped so this initialization is not needed.